pkg/scheduler/framework/plugins/imagelocality/image_locality.go

```diff
@@ -97,6 +97,10 @@ func calculatePriority(sumScores int64, numContainers int) int64 {
 func sumImageScores(nodeInfo *framework.NodeInfo, containers []v1.Container, totalNumNodes int) int64 {
 	var sum int64
 	for _, container := range containers {
+		// if pullPolicy is set to Always score it with 0 points
+		if container.ImagePullPolicy == "Always" {
```

Comment:
could this just fail earlier in the plugin, before needing to list all the nodes? if we know the container has an `Always`, then do we need to loop through the rest of the nodes? I suppose it will still have a scheduling effect for the other containers, so maybe we do need to do it this way. Just curious if it was considered

Reply:
True, I didnt consider it. We can have list of a containers which doesnt
have imagePullPolicy set to Always. Others will ve calculated normally and
one's with Always will either way always have 0 score so that doesnt matter.
I will update code later this day.